components/director/internal/domain/fetchrequest/service.go

```diff
@@ -75,38 +83,85 @@ func (s *service) fetchAPISpec(ctx context.Context, fr *model.FetchRequest) (*st
 	if resp.StatusCode != http.StatusOK {
 		errMsg := fmt.Sprintf("While fetching API Spec status code: %d", resp.StatusCode)
 		log.C(ctx).Errorf(errMsg)
-		return nil, s.fixStatus(model.FetchRequestStatusConditionFailed, str.Ptr(fmt.Sprintf("While fetching API Spec status code: %d", resp.StatusCode)))
+		return nil, FixStatus(model.FetchRequestStatusConditionFailed, str.Ptr(fmt.Sprintf("While fetching API Spec status code: %d", resp.StatusCode)), s.timestampGen())
 	}
 
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		log.C(ctx).WithError(err).Errorf("An error has occurred while reading API Spec.")
-		return nil, s.fixStatus(model.FetchRequestStatusConditionFailed, str.Ptr(fmt.Sprintf("While reading API Spec: %s", err.Error())))
+		log.C(ctx).WithError(err).Errorf("An error has occurred while reading Spec.")
+		return nil, FixStatus(model.FetchRequestStatusConditionFailed, str.Ptr(fmt.Sprintf("While reading API Spec: %s", err.Error())), s.timestampGen())
 	}
 
 	spec := string(body)
-	return &spec, s.fixStatus(model.FetchRequestStatusConditionSucceeded, nil)
+	return &spec, FixStatus(model.FetchRequestStatusConditionSucceeded, nil, s.timestampGen())
 }
 
 func (s *service) validateFetchRequest(fr *model.FetchRequest) error {
 	if fr.Mode != model.FetchModeSingle {
 		return apperrors.NewInvalidDataError("Unsupported fetch mode: %s", fr.Mode)
 	}
 
-	if fr.Auth != nil {
-		return apperrors.NewInvalidDataError("Auth for Fetch Request was provided, currently it's unsupported")
-	}
-
 	if fr.Filter != nil {
 		return apperrors.NewInvalidDataError("Filter for Fetch Request was provided, currently it's unsupported")
 	}
+
 	return nil
 }
 
-func (s *service) fixStatus(condition model.FetchRequestStatusCondition, message *string) *model.FetchRequestStatus {
+func (s *service) requestWithCredentials(ctx context.Context, fr *model.FetchRequest) (*http.Response, error) {
+	if fr.Auth.Credential.Basic == nil && fr.Auth.Credential.Oauth == nil {
+		return nil, apperrors.NewInvalidDataError("Credentials not provided")
+	}
+
+	req, err := http.NewRequest(http.MethodGet, fr.URL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp *http.Response
+	if fr.Auth.Credential.Basic != nil {
+		req.SetBasicAuth(fr.Auth.Credential.Basic.Username, fr.Auth.Credential.Basic.Password)
+
+		resp, err = s.client.Do(req)
+
+		if err == nil && resp.StatusCode == http.StatusOK {
```

Comment:
If both Basic & OAuth credentials are set and the request with basic fails it will attempt to make a request with OAuth credentials. Not sure this is expected behaviour.

A more general question is do we want to forbid FetchRequests with authentication data provided for both methods. We should discuss it with the team if it has not been already.

Reply:
Additionally, it may be possible for err = nil & httpstatus!=OK; this would mean that the `requestWithCredentials` function would return `nil, nil` (if on OAuth credentials provided), which would break the calling function when it inspects the response variable.

Reply:
It is intentional that if the request with basic credentials fails there will be an atempt to make request with OAuth credentials if such are present. It seemed appropriate to try the second set of credentilas if the first is not good. Although I am not sure if it is possible in our scenarios the two types of credentials to be provided at the same time.

Reply:
In case of err=nil & httpstatus!=OK there are two cases. First if there are OAuth credentials provided, request with them will be made and the returned err and responce will be returned from the function. In the second case there aren't provided OAuth credentials so we are not gonig into the if statement's body so the err(in this case nil) and the responce(from the request wit basic credentials which has status code different from OK) will be returned. The resp variable is intentionaly declared before the if statement(row 121) so that the value returned from the request is not lost.

Reply:
Oh, yes, you're right. Seems like I mislooked that piece of code:)